spec/test_helper.rb

```diff
@@ -14,15 +14,39 @@
 require 'byebug'
 require 'pathname'
 require 'lhm'
+require 'active_record'
 
 $project = Pathname.new(File.dirname(__FILE__) + '/..').cleanpath
 $spec = $project.join('spec')
 $fixtures = $spec.join('fixtures')
 
 $db_name = 'test'
 
-require 'active_record'
-require 'mysql2'
+Database = Struct.new(:adapter, :client, :error_class, :timeout_error) do
+  def query(connection, sql)
+    results = connection.query(sql)
+    results = results.each_hash if adapter == "trilogy"
+    results
+  end
+end
+
+DATABASE =
+  case ENV['DATABASE_ADAPTER']
+  when 'trilogy'
+    require 'trilogy'
+
+    if ActiveRecord.version < ::Gem::Version.new('7.1.0')
+      require 'activerecord-trilogy-adapter'
+      require 'trilogy_adapter/connection'
+
+      ActiveRecord::Base.public_send :extend, TrilogyAdapter::Connection
```

Comment:
Is this the line that helps with the specific issue in https://github.com/Shopify/lhm/pull/152?

Reply:
Small explanation for the fix: `execute` returns a different result based on what adapter you're using (mysql2, built-in trilogy, trilogy adapter gem), but `select_all` and `select_one` always return the same result. `execute` should also only be used when you expect side-effects since it'll clear the query cache. The `select_` methods are more suitable for what we're doing in most cases.